### PR TITLE
[mobile] ci: migrate to Bazel credential helper

### PR DESCRIPTION
**Pull request by Yannic**
Still using GitHub tokens for auth.

### CODE REVIEW

**Review by jpsim — DISMISSED** [review]
+1 thanks for doing this, might need a main merge again.

**Comment by jpsim** [comment]
I don't have permissions to merge this with failing checks. Alyssa, can you help out with this?

/assign @alyssawilk

**Comment by alyssawilk** [comment]
I can force merge over failures but they don't look like the standard tidy errors which resolve on merge and threes other failing builds. We sure this will land cleanly without breaking main?

**Comment by Yannic** [comment]
This only touches the Envov mobile on EngFlow jobs which all still pass. It doesn't seem like there's a linter complaining about the specific changes or the shell script I added, so the failures look unrelated to me.

**Comment by alyssawilk** [comment]
/retest
looks like mobile coverage has failed 3 times and I don't consider it particularly flaky.  Can you take a look at that and confirm this is expected and will resolve?

**Comment by jpsim** [comment]
Maybe a new `main` merge would make the jobs pass? We merged some PRs successfully since this branch was last updated.

**Comment by Yannic** [comment]
Reverted the change for coverage, but was still broken: https://github.com/envoyproxy/envoy/actions/runs/4441090791/jobs/7795726136?pr=26110

**Comment by Yannic** [comment]
Found it! `GITHUB_TOKEN` was not set in this job

**Comment by alyssawilk** [comment]
https://github.com/envoyproxy/envoy/pull/26086 is pretty recent and passing so I think there's probably something to sort out here?

**Review by alyssawilk — APPROVED** [review]

File: mobile/.bazelrc [hunk 1]
```diff
@@ -144,11 +144,12 @@ build:coverage --instrumentation_filter="//library/common[/:]"
 # GITHUB_TOKEN auth) so we have to override it to 'false' here.
 build:remote-ci-common --config=remote
 build:remote-ci-common --auth_enabled=false
+build:remote-ci-common --remote_cache=grpcs://envoy.cluster.engflow.com
 build:remote-ci-common --remote_executor=grpcs://envoy.cluster.engflow.com
 build:remote-ci-common --bes_backend=grpcs://envoy.cluster.engflow.com/
 build:remote-ci-common --bes_results_url=https://envoy.cluster.engflow.com/invocation/
+build:remote-ci-common --experimental_credential_helper=%workspace%/bazel/engflow-bazel-credential-helper.sh
 build:remote-ci-common --bes_lifecycle_events
-build:remote-ci-common --bes_upload_mode=fully_async
```
Comment (jpsim) [inline mobile/.bazelrc:151]: Why this change?

Comment (Yannic) [reply mobile/.bazelrc:151]: Otherwise, Bazel won't wait for BES to be fully uploaded, which can lead to incomplete streams if using short-lived runners like GH actions because Bazel server will just get terminated.

**Review by alyssawilk — APPROVED** [review]
